# Introduction



## Initial Data Exploration

### Examining the Data Sets

Here is my initial examination of the data:

In [3]:
import pandas as pd
import numpy as np

species_info = pd.read_csv('species_info.csv')
observations = pd.read_csv('observations.csv')

species_info.head()

category                scientific_name  \
0   Mammal  Clethrionomys gapperi gapperi   
1   Mammal                      Bos bison   
2   Mammal                     Bos taurus   
3   Mammal                     Ovis aries   
4   Mammal                 Cervus elaphus   

                                        common_names conservation_status  
0                           Gapper's Red-Backed Vole                 NaN  
1                              American Bison, Bison                 NaN  
2  Aurochs, Aurochs, Domestic Cattle (Feral), Dom...                 NaN  
3  Domestic Sheep, Mouflon, Red Sheep, Sheep (Feral)                 NaN  
4                                      Wapiti Or Elk                 NaN

In [4]:
observations.head()

scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85

In [11]:
print(species_info.describe())
print(species_info.info())
print(species_info.category.unique())
print(species_info.conservation_status.unique())


              category    scientific_name        common_names  \
count             5824               5824                5824   
unique               7               5541                5504   
top     Vascular Plant  Castor canadensis  Brachythecium Moss   
freq              4470                  3                   7   

       conservation_status  
count                  191  
unique                   4  
top     Species of Concern  
freq                   161  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5824 entries, 0 to 5823
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   category             5824 non-null   object
 1   scientific_name      5824 non-null   object
 2   common_names         5824 non-null   object
 3   conservation_status  191 non-null    object
dtypes: object(4)
memory usage: 182.1+ KB
None
['Mammal' 'Bird' 'Reptile' 'Amphibian' 'Fish' 'Vascular Plant'
 'Nonvascular Plant

In [16]:
print(observations.describe(include = 'all'))
print(observations.info())
print(observations.park_name.unique())

         scientific_name                            park_name  observations
count              23296                                23296  23296.000000
unique              5541                                    4           NaN
top     Myotis lucifugus  Great Smoky Mountains National Park           NaN
freq                  12                                 5824           NaN
mean                 NaN                                  NaN    142.287904
std                  NaN                                  NaN     69.890532
min                  NaN                                  NaN      9.000000
25%                  NaN                                  NaN     86.000000
50%                  NaN                                  NaN    124.000000
75%                  NaN                                  NaN    195.000000
max                  NaN                                  NaN    321.000000
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23296 entries, 0 to 23295
Data columns

### Thoughts on the Initial Data Exploration

Species_info lists 5541 unique species, along with their category (of which there are 7 categories), their common names and their conservation status. The conservation status is only on 191 different species and there are only 4 unique values.

Observation lists 23296 data points, which is exactly 4x the 5824 entries found in species_info. The data lists four unique entries under park_name, so my assumption is that each species has been observed in each park. The data also lists the number of observations of each species in each park.

### Potential Data Cleaning Up

There seem to be more data entries in the species column (5824) than there are species (5541), so there may be duplicates. I will need to check for this first.

The conservation status of every species other than the 191 listed likely needs to change to something like "No Issues." This will aid in some analysis.

There may also be duplicates in observations as it is exactly four times the size of species_info. This will need to be checked.

It may also prove wise to join the data in the two different data sets, though that may have to happen a bit later in the process.

## Questions for Exploration

Several questions arise from my initial exploration of the data:

- What are the proportions of different categories of animal types? 
- What are the proportions of animals at each conservation status?
- Are different categories of animals more or less likely to have a different conservations status? Eg Are more vascular plants endangered compared to fish?
- Which parks have more observations?
- What is the average observation for each species at each level of conservation status?

